### PR TITLE
Implement support for upstream kernel "set_flags".

### PR DESCRIPTION
**Pull request by ossama-othman**

### CODE REVIEW

File: src/netlink_pm_upstream.c [hunk 1]
```diff
@@ -624,6 +624,58 @@ static int upstream_get_limits(struct mptcpd_pm *pm,
...
+              Local address
+              Flags
+         */
+
+        // Types chosen to match MPTCP genl API.
+        uint16_t const family = mptcpd_get_addr_family(addr);
+
+        size_t const payload_size =
+                MPTCPD_NLA_ALIGN(uint16_t) // family
+                + NLA_HDRLEN + NLA_ALIGN(mptcpd_get_addr_size(addr))
```
Comment (mjmartineau) [inline src/netlink_pm_upstream.c:643]: I was wondering why this line was different, but then saw the `sizeof()` built in to `MPTCPD_NLA_ALIGN`. I recall you mentioning some kind of `sizeof`-related bug yesterday, is that the `MPTCPD_NLA_ALIGN(addr_size)` in `upstream_add_addr()`?

**Review by mjmartineau — COMMENTED** [review]
Looking good at this point

File: src/netlink_pm_upstream.c (hunk 1 above)
Comment (ossama-othman) [reply src/netlink_pm_upstream.c:643]: Right!  See PR #119 for my proposed fix.

Comment (ossama-othman) [reply src/netlink_pm_upstream.c]: I rebased this pull request to pull in the changes in #119.  This function now uses the `MPTCPD_NLA_ALIGN_ADDR()` macro.

**Review by mjmartineau — APPROVED** [review]
Looks good, thanks! Ran the self tests too.

**Comment by ossama-othman** [comment]
@mjmartineau Thanks for the review!

**Pull request merged**